### ISSUE TITLE
[clang-tidy] False positive: variable is not initialised with `::getpwuid`

### ISSUE THREAD

**Issue by H-G-Hristov** [llvm/llvm-project#57055]
The following snippet:

```c++
    const auto* data1 = ::getpwnam(userName.c_str());
```

produces:

    variable 'data1' is not initializedC/C++[cppcoreguidelines-init-variables](https://releases.llvm.org/14.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/cppcoreguidelines-init-variables.html)
    
Check tested with VSCode, C/C++ extension 1.11.5 on macOS Monterey.

![Screenshot 2022-08-10 at 18 49 22](https://user-images.githubusercontent.com/47526411/183952022-eb83bbeb-73c1-48f4-bc6b-c8baaf433aa3.png)

**Comment by EugeneZelenko** [comment]
Could you please try `main` branch? https://godbolt.org should be helpful.

**Comment by njames93** [comment]
Are there any other errors, These kind of issues can occur if the initializer is ill-formed.

**Comment by PiotrZSL** [comment]
Not reproduced, most probably caused by compilation errors.
Workaround looks to be added in ce6de98b80ada8efbff93766b5303db5ab052efc

**Comment by Zingam** [comment]
I continue seeing such false positives on macOS with std::vector, apple Darwin types,  all over the place.

**Comment by PiotrZSL** [comment]
Do you have any compilation errors ?
